src/include/sof/schedule/dp_schedule.h

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Szkudlinski
+ */
+
+#ifndef __SOF_SCHEDULE_DP_SCHEDULE_H__
+#define __SOF_SCHEDULE_DP_SCHEDULE_H__
+
+#include <sof/schedule/task.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <stdint.h>
+
+/**
+ *
+ * DP scheduler is a scheduler that creates a separate preemptible Zephyr thread for each SOF task
+ *   There's only one instance of DP in the system, however, the threads may run on any core
+ *   Threads are pinned to specific core, there's no SMP processing.
```

Comment:
Maybe easier to express "threads can be assigned to any core in the system, but each thread is pinned to a single core for its execution." Or maybe not, but you get the picture. So you can pick any core, but tasks stay on the core they were started on.

Reply:
> ATM this cannot work with a component on a core other than the pipeline. The problem is accessing buffer object list elements without locking in places like `pipeline_for_each_comp()`. We should block attempts to specify a core different from the pipeline core for DP components until such support is implemented.

I assure it can, at least works in tests now. The whole point of DP is to mix 1) diffrent ticks 2) diffrent cores